### PR TITLE
Add repair step to clean old calendar shares

### LINKED ISSUE #21889: [9.1] drop old calendar and contacts tables
With 9.0 we will migrate conatcts and calendar data over to the new core tables.
I don't want to run into the situation where we fail to execute the migration and delete the tables in 9.0.

Therefore we drop the tables with 9.1

clndr_calendars
clndr_objects
clndr_share_event
clndr_share_calendar
clndr_repeat

contacts_addressbooks
contacts_cards

### PR DESCRIPTION
**Pull request by rullzer**
fixes #21889

CC: @DeepDiver1975 @nickvergessen @PVince81

### CODE REVIEW

File: lib/private/Repair/RemoveOldCalendarShares.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
```
Comment (nickvergessen) [inline lib/private/Repair/RemoveOldCalendarShares.php]: missing header

**Comment by nickvergessen** [comment]
:+1:

**Comment by georgehrke** [comment]
We also had event shares back then.
Let me look up the item_type identifier

**Comment by georgehrke** [comment]
`event`
https://github.com/owncloudarchive/calendar/blob/master/appinfo/app.php#L30

**Comment by nickvergessen** [comment]
Contacts used `contact` and `addressbook` https://github.com/owncloudarchive/contacts/blob/master/appinfo/app.php#L45-L46

**Comment by rullzer** [comment]
Ok I can add those as well

**Comment by rullzer** [comment]
Added. please have another look

File: tests/lib/Repair/RemoveOldSharesTest.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ *
+ */
+namespace Test\Repair;
+
+use OC\Repair\RemoveOldShares;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+
+/**
+ * Class RemoveOldCalendarSharesTest
```
Comment (nickvergessen) [inline tests/lib/Repair/RemoveOldSharesTest.php]: :stuck_out_tongue:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by PVince81** [comment]
Code looks good :+1:

**Pull request merged**